src/amp-story-player/amp-story-player-impl.js

```diff
@@ -167,6 +183,7 @@ export class AmpStoryPlayer {
     this.element_.go = this.go.bind(this);
     this.element_.mute = this.mute.bind(this);
     this.element_.unmute = this.unmute.bind(this);
+    this.element_.getStoryState = this.getStoryState.bind(this);
```

Comment:
/cc @gmajoulet since this is a change to the public API

I think we should consider whether this is how we should expect people to interact with the underlying story state.  What's the use case here?  Do we need to expose this, or can it just be internal?

If we need to expose it, we should probably do more extensive API review

Reply:
We want to expose this so that publishers can get e.g. the page attachment state and perform corresponding actions on their end. (see [example](https://github.com/ampproject/amphtml/pull/29106#issuecomment-655006258))

Gabriel suggested opening this API here https://github.com/ampproject/amphtml/issues/29219#issuecomment-656769763 and here https://github.com/ampproject/amphtml/pull/29106#issuecomment-655755246

Can I ask what your concerns are @newmuis ?

Reply:
Friendly ping @newmuis @gmajoulet

Reply:
My concern was mostly just the level of abstraction.  Some things (e.g. play, pause) are actions you take on the player, even though they affect the current story.  Other things, you would need to check the exposed store service.

How do I know where to look?  Seems reasonable to me to have our developer-facing messaging be "always use the store", or "never use the store", but if our answer is "sometimes use the store", there should be a rationale about when to use it or when not to use it.

Reply:
What I had in mind when suggesting this getState/setState API is that we'd have:
1. MediaElement like API with play/pause/mute - which is cool and easy to use
2. getState/setState API to know anything (given we have good code + .md documentation) about the story

It duplicates the way you can mute/unmute a story, but (2) would also the only way to know if a story is muted or unmuted synchronously.

Reply:
Thanks @newmuis & @gmajoulet. I second Gabriel's thoughts and will echo we'll document this. 

Having this in mind, does the code in this PR look good?